src/app/messages/[id]/page.tsx

```diff
@@ -0,0 +1,33 @@
+import Conversation from '@/components/Conversation'
+import { Database } from '@/types/supabase'
+import { createServerComponentClient } from '@supabase/auth-helpers-nextjs'
+import { cookies } from 'next/headers'
+
+type Props = { params: { id: string } }
+
+export default async function Message({ params }: Props) {
+  const supabase = createServerComponentClient<Database>({
+    cookies,
+  })
+
+  const { data: profiles } = await supabase.from('profiles').select('*')
```

Comment:
suggestion: I still think that all these supabase queries should be outside React components. It should be hooks with selectors: useGetProfiles, useGetChat, useGetMessages, etc

Some benefits of it:
- Migrations: When the DB changes, you only need to change the hook itself instead of all the instances across the project.
- Cache: If you added react-query to the hook, then you benefit from cache across pages. For example `.from('profiles')` is done in 3 places that might be called multiple times in the app flow. With react-query it would be just 1 network request (depending on the cache config).
- Consistency: For example in profiles one of the places you called the variable `contacts` instead of `profiles`. With a hook, the naming is consistent.